app/helpers/quadicon_helper.rb

```diff
@@ -832,4 +831,31 @@ def quadicon_vt_link_options
 
     options
   end
+
+  #######################################
+  # Use OO Builders
+  #######################################
+
+  def quadicon_for(record, options = {})
+    context = Quadicons::Context.new(self) do |c|
+      # set attributes with view state
+      c.controller  = request.parameters[:controller]
+      c.edit        = @edit
+      c.embedded    = quadicon_in_embedded_view?
+      c.explorer    = quadicon_in_explorer_view?
+      c.lastaction  = @lastaction
+      c.listicon    = @listicon
+      c.listnav     = @listnav
+      c.listicon    = @listicon
```

Comment:
My current plan is the rest of the helper file would go away with only `quadicon_for` remaining. I might move this context setup code into the Context class.